src/transaction/deploy.rs

```diff
@@ -151,7 +152,13 @@ impl Deploy {
         state: &mut CachedState<S>,
         block_context: &BlockContext,
     ) -> Result<TransactionExecutionInfo, TransactionError> {
-        state.set_contract_class(&self.contract_hash, &self.contract_class)?;
+        match self.contract_class.clone() {
```

Comment:
I'm not sure this clone is necessary.

Reply:
Yeah it's not, removed here [47a2760](https://github.com/lambdaclass/starknet_in_rust/pull/943/commits/47a2760cb3187d9f05140254fe65b7799b3fa661)